teps/0081-add-chains-subcommand-to-the-cli.md

```diff
@@ -0,0 +1,337 @@
+dependency in order to implement the commands having to deal with the `cosign`
+keys and signatures.
+
+However, the Sigstore group is working on a lightweight version of `cosign` that
+could be more easily added as a dependency.
+
+### Risks and Mitigations
+
+<!--
+What are the risks of this proposal and how do we mitigate. Think broadly.
+For example, consider both security and how this will impact the larger
+kubernetes ecosystem.
+
+How will security be reviewed and by whom?
+
+How will UX be reviewed and by whom?
+
+Consider including folks that also work outside the WGs or subproject.
+-->
+
+### User Experience (optional)
+
+* Keep the CLI simple
+* Ensure auto-completion is available
+
+### Performance (optional)
+
+<!--
+Consideration about performance.
+What impact does this change have on the start-up time and execution time
+of task and pipeline runs? What impact does it have on the resource footprint
+of Tekton controllers as well as task and pipeline runs?
+
+Consider which use cases are impacted by this change and what are their
+performance requirements.
+-->
+
+## Design Details
+
+<!--
+This section should contain enough information that the specifics of your
+change are understandable.  This may include API specs (though not always
+required) or even code snippets.  If there's any ambiguity about HOW your
+proposal will be implemented, this is the place to discuss them.
+
+If it's helpful to include workflow diagrams or any other related images,
+add them under "/teps/images/". It's upto the TEP author to choose the name
+of the file, but general guidance is to include at least TEP number in the
+file name, for example, "/teps/images/NNNN-workflow.jpg".
+-->
+
+## Test Plan
+
+<!--
+**Note:** *Not required until targeted at a release.*
+
+Consider the following in developing a test plan for this enhancement:
+- Will there be e2e and integration tests, in addition to unit tests?
+- How will it be tested in isolation vs with other components?
+
+No need to outline all of the test cases, just the general strategy.  Anything
+that would count as tricky in the implementation and anything particularly
+challenging to test should be called out.
+
+All code is expected to have adequate tests (eventually with coverage
+expectations).
+-->
+
+* Unit tests should be enough.
+* The tests should be similar to any other tekton command.
+
+## Design Evaluation
+<!--
+How does this proposal affect the reusability, simplicity, flexibility
+and conformance of Tekton, as described in [design principles](https://github.com/tektoncd/community/blob/master/design-principles.md)
+-->
+
+## Drawbacks
+
+<!--
+Why should this TEP _not_ be implemented?
+-->
+
+## Alternatives
+
+The alternative is to use a combination of shell tools and to know which exact
+annotation or key to query/update. It was ruled out since it complicates
+the operations for no good reason. See the
+[Use Cases (optional)](use-cases-optional) section for some examples.
```

Comment:
There is one other alternative that would be that the `chains` provide a `tkn-chains` binary, and with the "execution model" we have in `tkn`, it would appear as a subcommand. One downside of this, is that it wouldn't be available by default and would complicate a bit the "packaging part". I'd rather have the chains command in, and secure by default (if history teach us anything is that things not enable or shipped by default are less adopted 😓 )

Reply:
+1!